godot-macros/src/docs.rs

```diff
@@ -135,13 +136,66 @@ fn xml_escape(value: String) -> String {
     result
 }
 
-/// Calls [`extract_docs_from_attributes`] and converts the result to BBCode for Godot's consumption.
-fn attribute_docs_to_bbcode(doc: &[venial::Attribute]) -> Option<String> {
-    let doc = extract_docs_from_attributes(doc)
-        .collect::<Vec<String>>()
-        .join("\n");
+/// Returns user docs with its `@deprecated` and `@experimental` attributes extracted.
+fn docs_with_attributes(doc: &[venial::Attribute]) -> (Vec<String>, Vec<String>, Vec<String>) {
+    let (mut docs, mut deprecated, mut experimental): (Vec<String>, Vec<String>, Vec<String>) =
+        (Vec::new(), Vec::new(), Vec::new());
```

Comment:
`docs_with_attributes` could be named `split_paragraphs`, and maybe it's worth having a dedicated struct?
```rs
#[derive(Default)]
struct Paragraphs {
    description: Vec<String>,
    deprecated: Vec<String>,
    experimental: Vec<String>,
}
```

Also, is there any point in having `Vec` at all here? If we directly append to 3 `String`s inside this function, we could save the `Vec` allocation.

As far as I see, there's just `piece.join("\n")` after this, anyway...